---
title: "Novel Peptides"
author: Marc Vaudel
date: 2019-10-28
output: 
    github_document:
    toc: true
    toc_depth: 1
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# 1. Novel Peptides

Mutations in the genome can alter the sequence so that non-coding sections become coding. _e.g._ through the introduction of a start codon or the alteration of a stop codon. In order to identify these non-canonical genomic products, protein databases that capture genetic variation and non-canonical genomic products are generated either by enriching canonical protein sequences or by running six reading frame translation of the entire genome [(1)](#references). 

##### [:thought_balloon:](answers.md#thought_balloon-based-on-you-knowledge-of-peptide-and-protein-identification-can-you-anticipate-challenges-posed-by-these-proteogenomic-databases) _Based on you knowledge of peptide and protein identification, can you anticipate challenges posed by these proteogenomic databases?_


## Processing

This tutorial is a notebook that contains [R](r-project.org) code that can be run directly from the _Rmd_ file. It assumes that the R working directory is the proteogenomics folder of the repository, _e.g._ `/myfolder/IBIP19/pages/proteogenomics`. We recommend using RStudio to run this tutorial.


## Libraries

You will need the following libraries, please make sure that they are installed.

```{r libraries, warning=FALSE, results='hide', message=FALSE}

library(dplyr)

```


## Data set

In this tutorial, we will analyze the non-canonical genomic products identified in breast cancer by Johansson _et al._ [(2)](#references). Note that this tutorial does not cover the database generation, search, and validation of identification results. These bioinformatic procedures are very demanding and we strongly advise to make sure that they are in place at your lab or at the facility processing the data before conducting any proteogenomic experiment. The proteogenomic identification results by Johansson _et al._ [(2)](#references) are reported in Supplementary Data 6, available [here](../resources/Johansson_et_al_breast_cancer_quantitative_proteome_and_proteogenomic_landscape) in the course repository.

##### [:thought_balloon:](answers.md#thought_balloon-what-do-the-different-columns-in-the-table-represent) _What do the different columns in the table represent?_

For this tutorial, the _Novel Peptides_ table was extracted to an R-friendly text format, and is available [here](resources/data/novel_peptides.txt).

##### :pencil2: Load the data in R as in the code below.

```{r import data}

novelPeptidesDF <- read.table(
    file = "resources/data/novel_peptides.txt",
    header = T,
    sep = "\t",
    comment.char = "",
    stringsAsFactors = F
)

```

##### :pencil2: Find the different classes of loci represented.

```{r loci}

knitr::kable(
    as.data.frame(
        table(
            novelPeptidesDF$class
            )
        )
    )

```

##### [:thought_balloon:](answers.md#thought_balloon-based-on-you-knowledge-of-peptide-and-protein-identification-can-you-anticipate-challenges-posed-by-these-proteogenomic-databases) _How can we speculate on these different classes of loci yielding detectable peptides?_

## References

(1) [Proteogenomics: concepts, applications and computational strategies](https://www.ncbi.nlm.nih.gov/pubmed/25357241)

(2) [Breast cancer quantitative proteome and proteogenomic landscape](https://www.ncbi.nlm.nih.gov/pubmed/30962452)

