---
title: "Novel Peptides"
author: Marc Vaudel
date: 2019-10-28
output: 
    github_document:
    toc: true
    toc_depth: 1
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# Novel Peptides

Mutations in the genome can alter the sequence so that non-coding sections become coding. _e.g._ through the introduction of a start codon or the alteration of a stop codon. In order to identify these non-canonical genomic products, protein databases that capture genetic variation and non-canonical genomic products are generated either by enriching canonical protein sequences or by running six reading frame translation of the entire genome [(1)](#references). 

##### [:thought_balloon:](answers.md#thought_balloon-based-on-you-knowledge-of-peptide-and-protein-identification-can-you-anticipate-challenges-posed-by-these-proteogenomic-databases) _Based on you knowledge of peptide and protein identification, can you anticipate challenges posed by these proteogenomic databases?_


## Libraries

We will need the following libraries, please make sure that they are installed.

```{r libraries, warning=FALSE, results='hide', message=FALSE}

library(tidyr)
library(dplyr)
library(ggplot2)
library(scico)

theme_set(theme_bw(base_size = 11))

```


## Data set

In this tutorial, we will analyze the non-canonical genomic products identified in breast cancer by Johansson _et al._ [(2)](#references). Note that this tutorial does not cover the database generation, search, and validation of identification results. These bioinformatic procedures are very demanding and we strongly advise to make sure that they are in place at your lab or at the facility processing the data before conducting any proteogenomic experiment. The proteogenomic identification results by Johansson _et al._ [(2)](#references) are reported in Supplementary Data 6, available [here](../resources/Johansson_et_al_breast_cancer_quantitative_proteome_and_proteogenomic_landscape) in the course repository.

##### [:thought_balloon:](answers.md#thought_balloon-what-do-the-different-columns-in-the-table-represent) _What do the different columns in the table represent?_

For this tutorial, the _Novel Peptides_ table was extracted to an R-friendly text format, and is available in [resources/data/novel_peptides.gz](resources/data/novel_peptides.gz).

##### :pencil2: Load the data in R as in the code below.

```{r import_id}

novelPeptidesDF <- read.table(
    file = "resources/data/novel_peptides.gz",
    header = T,
    sep = "\t",
    comment.char = "",
    quote = "",
    stringsAsFactors = F
)

```


## Genomic context and function

##### :pencil2: Find the different classes of loci represented.

```{r loci_class}

classesDF <- as.data.frame(
    table(
        novelPeptidesDF$class
    )
) %>%
    rename(
        class = Var1,
        n_peptides = Freq
    ) %>%
    arrange(
        desc(n_peptides)
    )

knitr::kable(classesDF)


```

##### [:thought_balloon:](answers.md#thought_balloon-can-you-speculate-on-how-these-different-classes-of-loci-can-yield-novel-peptides) _Can you speculate on how these different classes of loci can yield novel peptides?_

##### :pencil2: For the non-intergenic peptides, find the different categories of associated genes.

```{r gene_categories}

novelPeptidesDF %>% 
    filter(
        class != "intergenic"
    ) %>%
    select(
        nearest_gene, category
    ) -> geneDF

categoriesDF <- as.data.frame(
    table(
        geneDF$category
    )
) %>%
    rename(
        category = Var1,
        n_peptides = Freq
    ) %>%
    arrange(
        desc(n_peptides)
    )

knitr::kable(categoriesDF)

```

##### [:thought_balloon:](answers.md#thought_balloon-what-do-these-categories-represent) _What do these categories represent?_

##### :pencil2: Select peptides from the different classes and categories, and inspect the genetic landscape at these positions using the Ensembl or UCSC genome browsers. [Example for locus #9 (Intronic)](https://grch37.ensembl.org/Homo_sapiens/Location/View?r=1:51913622-51913666)

##### :pencil2: Select genes possibly influenced by the transcription/translation change and inspect the function of the associated proteins.

## Abundance in tumors and normal tissue

In [Supplementary Table 8](../resources/Johansson_et_al_breast_cancer_quantitative_proteome_and_proteogenomic_landscape), the authors provide the abundance for novel peptides monitored in normal tissue and tumors for five patients. The table was extracted to an R-friendly text format for this tutorial, and is available in [resources/data/novel_peptides_paired.gz](resources/data/novel_peptides_paired.gz).

##### :pencil2: Load the data in R as done for the previous table. In addition, transform the data from wide to long format and create columns indicating the patient number, whether the sample is _Control_ or _Tumor_, and what kind of tumor, as done in the code below.

```{r import_quant}

novelPeptidesDF <- read.table(
    file = "resources/data/novel_peptides_paired.gz",
    header = T,
    sep = "\t",
    comment.char = "",
    quote = "",
    stringsAsFactors = F
) %>% 
    gather(
        "Control_1", "Control_2", "Control_3", "Control_4", "Control_5",
        key = "control_id",
        value = "control"
    ) %>% 
    select(
        -control_id
    ) %>%
    gather(
        "LumA_1", "Her2_2", "LumB_3", "Basal_4", "Her2_5",
        key = "tumor_id",
        value = "tumor"
    ) %>%
    separate(
        col = "tumor_id",
        into = c("tumorType", "patientNumber"),
        sep = "_"
    ) %>%
    mutate(
        patientId = paste("Patient", patientNumber)
    ) %>%
    arrange(
        abs(tumor - control)
    )

```

##### :pencil2: Plot the peptide abundance in the tumor _vs_ control tissue for all peptides and all patients.

```{r facet_patient}



ggplot(
    data = novelPeptidesDF
) +
    geom_hline(
        yintercept = quantile(
            x = novelPeptidesDF$tumor, 
            probs = c(0.2, 0.8),
            na.rm = T
        ),
        col = "black",
        linetype = "dotted"
    ) +
    geom_vline(
        xintercept = quantile(
            x = novelPeptidesDF$control, 
            probs = c(0.2, 0.8),
            na.rm = T
        ),
        col = "black",
        linetype = "dotted"
    ) +
    geom_point(
        mapping = aes(
            x = control,
            y = tumor,
            col = log10(ms1_area)
        )
    ) +
    facet_grid(
        tumorType ~ .
    ) +
    scale_x_log10(
        name = "Intensity in control tissue"
    ) +
    scale_y_log10(
        name = "Intensity in tumor tissue"
    ) + 
    scale_color_scico(
        name = "MS1 Area [log10]",
        palette = "batlow"
    ) +
    theme(
        legend.position = "top",
        panel.grid = element_blank()
    )

````

##### :speech_balloon: How do you interpret this plot?


## Conclusion

##### :speech_balloon: Can you speculate on the function or effect of these novel peptides in cancer biology? How can these be used in a clinical setup?


## References

(1) [Proteogenomics: concepts, applications and computational strategies](https://www.ncbi.nlm.nih.gov/pubmed/25357241)

(2) [Breast cancer quantitative proteome and proteogenomic landscape](https://www.ncbi.nlm.nih.gov/pubmed/30962452)

