---
title: "Variation Analysis"
author: Marc Vaudel
date: 2019-10-28
output: 
    github_document:
    toc: true
    toc_depth: 1
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# 1. Variation Analysis

In the [previous chapter](novel_peptides.md), novel peptides were identified by Johansson _et al._ [(1)](#references) using a proteogenomic search strategy. In addition, this search allowed identifying peptides presenting single amino acid variants (SAAV). 


## Libraries

We will need the following libraries, please make sure that they are installed.

```{r libraries, warning=FALSE, results='hide', message=FALSE}

library(tidyr)
library(dplyr)
library(ggplot2)
library(scico)

theme_set(theme_bw(base_size = 11))

```


## Genotype - peptide abundance relationship

Figure 7h displays the intensity of peptides produced by the reference and alternative alleles of corresponding SNPs for each genotype.

![Figure_7h](resources/images/Fig7h.png?raw=true "Johansson et al. Fig 7h")

> Figure 7h in Johansson _et al._ [(1)](#references).

##### [:thought_balloon:](answers.md#thought_balloon-if-we-assume-a-linear-relationship-between-number-of-alleles-and-peptide-abundance-what-should-be-the-peptide-distribution-for-each-genotype) _If we assume a linear relationship between number of alleles and peptide abundance, what should be the peptide distribution for each genotype?_

Note that the genotyping data corresponding to the peptide intensities were not made available, but more examples can be found in [Supplementary Figure 15](../resources/Johansson_et_al_breast_cancer_quantitative_proteome_and_proteogenomic_landscape/supplementary_information.pdf).

##### :speech_balloon: Do the results presented follow a linear trend? What can affect the linearity of the relationship between number of alleles and intensity distribution? 


## Abundance of variant peptides relative to the proteins

In [Supplementary Table 1](../resources/Johansson_et_al_breast_cancer_quantitative_proteome_and_proteogenomic_landscape) and [Supplementary Table 6](../resources/Johansson_et_al_breast_cancer_quantitative_proteome_and_proteogenomic_landscape), the authors provide the results of their proteogenomic analysis for genes and for peptides carrying alternative alleles of SAAVs, respectively. In addition, [Supplementary Table 1](../resources/Johansson_et_al_breast_cancer_quantitative_proteome_and_proteogenomic_landscape) contains experimental design and sample characterization information. The tables were extracted to an R-friendly text format for this tutorial, and are available in [resources/data/tumor.gz](resources/data/tumor.gz), [resources/data/proteins.gz](resources/data/proteins.gz), and [resources/data/saav.gz](resources/data/saav.gz).

##### :pencil2: Load the data in R as in the code below.

```{r import}

tumorDF <- read.table(
    file = "resources/data/tumor.gz",
    header = T,
    sep = "\t",
    comment.char = "",
    quote = "",
    stringsAsFactors = F
)

proteinsDF <- read.table(
    file = "resources/data/proteins.gz",
    header = T,
    sep = "\t",
    comment.char = "",
    quote = "",
    stringsAsFactors = F
) %>%
    select(
        -contains("nspsm")
    )

saavDF <- read.table(
    file = "resources/data/saav.gz",
    header = T,
    sep = "\t",
    comment.char = "",
    quote = "",
    stringsAsFactors = F
)

```

Note that for the sake of speed, the last columns of the protein table were skipped.

##### :pencil2: Normalize the intensities of the SAAV peptides using the reference channel (131).

```{r import}

for (tmtColumn in c("tmt126", "tmt127N", "tmt127C", "tmt128N", "tmt128C", "tmt129N", "tmt129C", "tmt130N", "tmt130C", "tmt131")) {
    
    saavDF <- saavDF %>% 
        mutate(
            !!tmtColumn := !!sym(tmtColumn) / tmt131
        )
}

saavDF <- saavDF %>%
    select(
        -tmt131
    )

```

As you might have noticed, the quantitative values in the protein table have been arranged by tumors, but this is not the case for SAAV peptides.

##### :pencil2: Organize the SAAV peptides table using kits as column, keep one quantification value per peptide.

```{r organize_saav}

# Get the unique peptides
saavPeptidesProteinDF <- saavDF %>% 
    select(
        peptide_sequence,
        protein
    ) %>%
    distinct()

# Iterate all sets
for (tmtSet in 1:5) {
    
    # Set up intensity columns
    for (tmtColumn in c("tmt126", "tmt127N", "tmt127C", "tmt128N", "tmt128C", "tmt129N", "tmt129C", "tmt130N", "tmt130C")) {
        
        newColumn <- paste0(tmtColumn, "_set", tmtSet)
        saavPeptidesProteinDF[[newColumn]] <- NA
        
    }
    
    # Gather intensities for each set and each peptide 
    for (i in 1:nrow(saavPeptidesProteinDF)) {
        
        js <- saavDF$tmt_set == tmtSet & saavDF$peptide_sequence == saavPeptidesProteinDF$peptide_sequence[i]
        
        if (sum(js) > 0) {
            
            for (tmtColumn in c("tmt126", "tmt127N", "tmt127C", "tmt128N", "tmt128C", "tmt129N", "tmt129C", "tmt130N", "tmt130C")) {
                
                newColumn <- paste0(tmtColumn, "_set", tmtSet)
                
                values <- saavDF[js, tmtColumn]
                
                if (sum(!is.na(values)) > 0) {
                    
                    value <- median(
                        x = values, 
                        na.rm = T
                    )
                    
                    saavPeptidesProteinDF[i, newColumn] <- value
                    
                }
            }
        }
    }
}

```

##### :pencil2: Use the experimental design to label the columns according to the tumors analyzed.

```{r rename_columns}

tumorDF <- tumorDF %>%
    mutate(
        tmtColumn = paste0("tmt", tmtTag, "_set", tmtSet)
    )

for (i in 1:nrow(tumorDF)) {
    
    oldColumn <- tumorDF$tmtColumn[i]
    newColumn <- tumorDF$tumor_id[i]
    
    names(saavPeptidesProteinDF)[names(saavPeptidesProteinDF) == oldColumn] <- newColumn
    names(proteinsDF)[names(proteinsDF) == oldColumn] <- newColumn
    
}


```

##### :pencil2: Extract gene names and merge with the protein table.

```{r rename_columns}

mergedDF <- NULL

genesSplit <- strsplit(
    x = saavPeptidesProteinDF$protein,
    split = ";"
)

for (i in 1:nrow(saavPeptidesProteinDF)) {
    
    geneLines <- genesSplit[[i]]
    
    geneLines <- geneLines[
        startsWith(
            x = geneLines,
            "CanProVar"
        ) |
            startsWith(
                x = geneLines,
                "COSMIC"
            ) |
            startsWith(
                x = geneLines,
                "RNAediting"
            )
        ]
    
    peptideDF <- data.frame(
        gene = character(length(geneLines)),
        peptide_sequence = character(length(geneLines)),
        gene_variant = character(length(geneLines)),
        protein_variant = character(length(geneLines)),
        stringsAsFactors = F
    )
    
    for (j in 1:length(geneLines)) {
        
        geneLine <- geneLines[1]
        
        gene <- NA
        geneVariant <- NA
        proteinVariant <- NA
        
        if (
            startsWith(
                x = geneLine,
                "CanProVar"
            )) {
            
            tempSplit <- strsplit(
                x = geneLine,
                split = "_"
            )[[1]]
            
            gene <- tempSplit[3]
            geneVariant <- tempSplit[2]
            proteinVariant <- tempSplit[4]
            
        } else if (
            startsWith(
                x = geneLine,
                "COSMIC"
            )) {
            
            tempSplit <- strsplit(
                x = geneLine,
                split = ":"
            )[[1]]
            
            gene <- tempSplit[2]
            geneVariant <- strsplit(
                x = tempSplit[4],
                split = "."
            )[[1]][2]
            proteinVariant <- strsplit(
                x = tempSplit[5],
                split = "."
            )[[1]][2]
            
            tempSplit <- strsplit(
                x = gene,
                split = "_"
            )[[1]]
            
            if (length(tempSplit) > 1) {
                
                gene <- tempSplit[1]
                
            }
        } else if (
            startsWith(
                x = geneLine,
                "RNAediting"
            )) {
            
            tempSplit <- strsplit(
                x = geneLine,
                split = "_"
            )[[1]]
            
            tempSplit2 <- strsplit(
                x = tempSplit[7],
                split = "."
            )[[1]]
            
            gene <- tempSplit[3]
            geneVariant <- paste(tempSplit[8], tempSplit[9], tempSplit[2], sep = ".")
            proteinVariant <- paste(tempSplit2[1], tempSplit[6], tempSplit2[2], sep = ".")
            
        }
        
        peptideDF$peptide_sequence[j] <- saavPeptidesProteinDF$peptide_sequence[i]
        peptideDF$gene[j] <- gene
        peptideDF$gene_variant[j] <- geneVariant
        peptideDF$protein_variant[j] <- proteinVariant
        
        for (tumor in tumorDF$tumor_id) {
            
            peptideColumn <- paste0(tumor, "_peptide")
            peptideValue <- saavPeptidesProteinDF[i, tumor]
            peptideDF[j, peptideColumn] <- peptideValue
            
            proteinColumn <- paste0(tumor, "_protein")
            
            k <- proteinsDF$gene_symbol == peptideDF$gene[j]
            
            if (sum(k) == 1) {
                
                proteinValue <- proteinsDF[k, tumor]
                
            } else {
                
                proteinValue <- NA
                
            }
            
            peptideDF[j, proteinColumn] <- proteinValue
            
            peptideDF[j, tumor] <- peptideValue / proteinValue
            
        }
    }
    
    peptideDF <- peptideDF %>%
        distinct()
    
    if (is.null(mergedDF)) {
        
        mergedDF <- peptideDF
        
    } else {
        
        mergedDF <- rbind(mergedDF, peptideDF)
        
    }
}


```

## References

(1) [Breast cancer quantitative proteome and proteogenomic landscape](https://www.ncbi.nlm.nih.gov/pubmed/30962452)
