---
layout: page
title: IBIP19
tagline: Integrative Biological Interpretation using Proteomics
description: This training course is aimed at researchers who are not expert in proteomics and want to integrate quantitative proteomics results into wider biomedical experiments. 
---

## Course Description

Quantitative proteomics by mass spectrometry has become an essential tool for multi-omics studies aiming at answering important biological questions in a system-wide perspective. Proteomics data contain rich and deep information that can become challenging to extract, interpret, and interface with other experimental outcomes.
This training course is aimed at researchers who are not expert in proteomics and want to integrate quantitative proteomics results into wider biomedical experiments. We will focus on quality control from an end-user perspective, link to the underlying genomic context, multivariate analysis, protein complexes investigation, and compare different platforms for biological interpretation.
After the training, participants will be able to critically interpret results, troubleshoot analyses, and will be ready to successfully attend more specialized training _e.g._ in proteogenomics or biological network analysis. 


# Prerequisites

Strictly speaking, there are no pre-requisites. Prior knowledge that will be useful, but not required: 
- Acquaintance with bottom-up proteomics workflow (sample preparation, LC-MS, terminology)
- Basics of proteomics identification (databases, search engines, protein inference)
- Basics of quantitative proteomics (labeling vs. label free, bioinformatics methods to infer peptide and protein abundances)
- Awareness of protein inference issues in proteomics identification and quantification
- Basics of scripting in R/Python


## Preliminary Program

### [Day 1](pages/day1.md): using proteomics data
- What is proteomics good for? What kind of questions can and cannot be answered. What is the interface/complementarity with other omics.
- I have received results from a core facility - where do I start? How do I visualize and QC the data?
- I donâ€™t find what I was expecting, what could have gone wrong?

### [Day 2](pages/day2.md): navigating quantitative proteomics data
- What are the most important variables in my analysis?
- How do I handle covariates and batch effects?
- What are the quantitative clusters in my experiment?

### [Day 3](pages/day3.md): from quantities to biological insights
- Are these clusters linked to specific pathways/biological processes?
- Quantitative profiling of protein complexes
- Comparison of different platforms for functional enrichment of biological pathways and their network representation

### [Day 4](pages/proteogenomics.md): back to the genome
- How do I map my proteomics results to the genome?
- How do I find the effect of genomic variation to the genome?
- How do I match genomics, transcriptomics, and proteomics data and what to expect from the results?


---

### Learning objectives

### [Instructors](pages/instructors.md)


---

The source for this course webpage is [on github](https://github.com/GTPB/Web_course_template).
