---
layout: page
title: Proteogenomics
---

# Proteogenomics

Proteogenomics is the scientific field at the interface between proteomics and genomics [(1)](#references). First, proteogenomic applications focused on improving genomic annotation through the use of proteomics data [(2)](#references). Over the years, proteogenomics rapidly developed as a rich scientific field for the joint analysis of genomic and proteomic information, generating fundamental knowledge on the interplay between genome and proteome.

Due to the novelty of the field, the complexity of the analyses, and the volume of data involved, proteogenomics poses multiple bioinformatic challenges [(3)](#references). These tutorials present methods that can be used to analyse results from proteogenomic pipelines. 

## A word of caution

Please note that this tutorial starts from the results of proteomic and genomic analyses, and does not cover the details of respective bioinformatic pipelines, like the calling of variants. Please note also that the generation and analysis of proteomic data for proteogenomic application differ from standard proteomic analyses, requiring specific methods and expertise. Notably, the identification of alternative products of transcription/translation and the calling of variants in proteomics data require high coverage and high resolution proteomic analyses. These analyses also require the use of complex bioinformatic workflows in high performance environments. Finally, they can benefit from tailoring the experimental workflow in relationship to the bioinformatic analysis [(4)](#references).

Note that establishing a proteogenomic workflow, in both wet- and dry-lab is a long-term endeavor. If you plan to conduct a proteogenomic experiment, please give particular attention to the following questions:
- Do I have enough sample material, replicates, and controls?
- Has the proteomics lab sufficient instrumentation and expertise to run the samples?
- Is the proteogenomic bioinformatic pipeline in place and able to handle the volume of data generated by my experiment?

When in doubt, consider running a small pilot experiment to benchmark the performance of your analytical set-up. Also, prioritize proteomics labs that have experience with proteogenomic analyses and analyzing the data.

## Tutorials

- [Novel peptides](): mapping proteomics results to non-coding loci.

- [Variation Analysis](): studying sequence and splicing variation in proteomics data.

- [CNA-Protein](): linking structural variants and protein levels to study CNA dampening/silencing.

- [RNA-Protein](): compatring RNA and protein levels to identify key biological mechanisms.


## References

(1) [Proteogenomics: concepts, applications and computational strategies](https://www.ncbi.nlm.nih.gov/pubmed/25357241)
(2) [Proteogenomic mapping as a complementary method to perform genome annotation](https://www.ncbi.nlm.nih.gov/pubmed/14730672)
(3) [Proteogenomics from a bioinformatics angle: A growing field](https://www.ncbi.nlm.nih.gov/pubmed/26670565)
(4) [HiRIEF LC-MS enables deep proteome coverage and unbiased proteogenomics](https://www.ncbi.nlm.nih.gov/pubmed/24240322)
